tools/ci_credentials/ci_credentials/secrets_loader.py

```diff
@@ -0,0 +1,240 @@
+        if self.connector_name:
+            filter += f" AND labels.connector={self.connector_name}"
+        url = f"https://secretmanager.googleapis.com/v1/projects/{self.api.project_id}/secrets"
+        next_token = None
+        while True:
+            params = {
+                "filter": filter,
+            }
+            if next_token:
+                params["pageToken"] = next_token
+            data, err = self.api.get(url, params=params)
+            if err:
+                return self.logger.critical(f"Google list error: {err}")
+
+            for secret_info in data.get("secrets") or []:
+                secret_name = secret_info["name"]
+                connector_name = secret_info.get("labels", {}).get("connector")
+                if not connector_name:
+                    self.logger.warning(f"secret {secret_name} doesn't have the label 'connector'")
+                    continue
+                elif self.connector_name and connector_name != self.connector_name:
+                    self.logger.warning(f"incorrect the label connector '{connector_name}' of secret {secret_name}")
+                    continue
+                filename = secret_info.get("labels", {}).get("filename")
+                if filename:
+                    # all secret file names should be finished with ".json"
+                    # but '.' cant be used in google, so we append it
+                    filename = f"{filename}.json"
+                else:
+                    # the "filename" label is optional.
+                    filename = DEFAULT_SECRET_FILE_WITH_EXT
+                log_name = f'{secret_name.split("/")[-1]}({connector_name})'
+                self.logger.info(f"found GSM secret: {log_name} = > {filename}")
+                secret_url = f"https://secretmanager.googleapis.com/v1/{secret_name}/versions/latest:access"
+
+                data, err = self.api.get(secret_url)
+                if err:
+                    return self.logger.critical(f"Google secret's error {secret_url} => {err}")
+                secret_value = data.get("payload", {}).get("data")
+                if not secret_value:
+                    self.logger.warning(f"{log_name} has empty value")
+                    continue
+                secret_value = base64.b64decode(secret_value.encode()).decode('utf-8')
+                try:
+                    # minimize and validate its JSON value
+                    secret_value = json.dumps(json.loads(secret_value), separators=(',', ':'))
+                except JSONDecodeError as err:
+                    self.logger.error(f"{log_name} has non-JSON value!!! Error: {err}")
+                    continue
+                secrets[(connector_name, filename)] = secret_value
+            next_token = data.get("nextPageToken")
+            if not next_token:
+                break
+        return secrets
+
+    @classmethod
+    def generate_secret_name(cls, connector_name: str, filename: str) -> str:
```

Comment:
this looks like a staticmethod rather than classmethod

Reply:
agree, changed